staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -111,6 +112,47 @@ func (mounter *Mounter) hasSystemd() bool {
 	return *mounter.withSystemd
 }
 
+// Do a bind mount including the needed remount for applying the bind opts.
+// If the remount fails and we are running in a user namespace
+// figure out if the source filesystem has the ro, nodev, noexec, nosuid,
+// noatime, relatime or nodiratime flag set and try another remount with the found flags.
+func (mounter *Mounter) bindMountSensitive(mounterPath string, mountCmd string, source string, target string, fstype string, bindOpts []string, bindRemountOpts []string, bindRemountOptsSensitive []string, mountFlags []string, systemdMountRequired bool) error {
+	err := mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if err != nil {
+		return err
+	}
+	err = mounter.doMount(mounterPath, defaultMountCommand, source, target, fstype, bindRemountOpts, bindRemountOptsSensitive, mountFlags, systemdMountRequired)
+	if libcontaineruserns.RunningInUserNS() {
```

Comment:
`k8s.io/kubernetes/pkg/kubelet/userns/inuserns.RunningInUserNS` can be used

Reply:
@AkihiroSuda: This does not seem to work for things below staging until the code from 2e999fff02650b07f0e4ffb9c4563548c669b9c3 is in a release. Currently it is only part of [v1.30.0-alpha.2](https://github.com/kubernetes/kubernetes/releases/tag/v1.30.0-alpha.2) which is marked as pre-release.
Hence `hack/verify-vendor.sh`  currently errors out with

```
=== computing imports for k8s.io/mount-utils
+ go list all
go: finding module for package k8s.io/kubernetes/pkg/kubelet/userns/inuserns
go: downloading k8s.io/kubernetes v1.29.1
module k8s.io/kubernetes@latest found (v1.29.1), but does not contain package k8s.io/kubernetes/pkg/kubelet/userns/inuserns

```
Any suggestions?

Reply:
This PR will be merged after v1.30 is released (it's freeze now). Do you plan to backport it to older releases?

Reply:
I do not plan to backport it. Hence I will move forward once `hack/verify-vendor.sh`stops failing.

Reply:
I think code in staging repos cannot import `k8s.io/kubernetes`. Using `github.com/opencontainers/runc/libcontainer/userns` is IMO correct here.